src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -145,13 +145,20 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
-     * return the entity instance of the citizen data.
+     * Return the entity instance of the citizen data. Respawn the citizen if needed.
      *
      * @return {@link EntityCitizen} of the citizen data.
      */
+    //TODO evaluate if this is nullable. Will spawnCitizen fail?
     @Nullable
```

Comment:
`    public void spawnCitizen(final CitizenData data)
    {
        final BlockPos townHallLocation = townHall.getLocation();
        if (!world.isBlockLoaded(townHallLocation))
        {
            //  Chunk with TownHall Block is not loaded
            return;
        }`

We potentially don't respawn